core/key.go

```diff
@@ -67,6 +68,15 @@ type DataStoreKey struct {
 
 var _ Key = (*DataStoreKey)(nil)
 
+// IndexDataStoreKey is a type that represents a key of an indexed document in the database.
+type IndexDataStoreKey struct {
+	CollectionID string
```

Comment:
todo: It is non-obvious to me as to which collection id this is.  Can you please document whether it is the integer (as a string), the cid, or the name?

suggestion: `IndexID` is less ambiguous, but I think it is worth being really explicit and add some documentation for the prop.